misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -0,0 +1,234 @@
+        Position startPos = new Position(bLangNode.pos.sLine - 1, bLangNode.pos.sCol - 1);
+        Position endPosWithSemiColon = new Position(bLangNode.pos.eLine - 1, bLangNode.pos.eCol);
+        Range newTextRange = new Range(startPos, endPosWithSemiColon);
+
+        BSymbol symbolAtCursor = refAtCursor.getSymbol();
+        String generatedFunctionName =
+                String.format("map%sTo%s(%s);", foundTypeRight, foundTypeLeft, symbolAtCursor.name.value);
+        fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
+
+        // Insert function declaration at the bottom of the file
+        WorkspaceDocumentManager docManager = context.get(DocumentServiceKeys.DOC_MANAGER_KEY);
+        Optional<Path> filePath = CommonUtil.getPathFromURI(context.get(DocumentServiceKeys.FILE_URI_KEY));
+        if (filePath.isPresent()) {
+            String fileContent = docManager.getFileContent(Paths.get(String.valueOf(filePath.get())));
+            String functionName = String.format("map%sTo%s (%s", foundTypeRight, foundTypeLeft, foundTypeRight);
+            if (!fileContent.contains(functionName)) {
+                int numberOfLinesInFile = fileContent.split("\n").length;
+                Position startPosOfLastLine = new Position(numberOfLinesInFile + 2, 0);
+                Position endPosOfLastLine = new Position(numberOfLinesInFile + 2, 1);
+                Range newFunctionRange = new Range(startPosOfLastLine, endPosOfLastLine);
+                String generatedRecordMappingFunction =
+                        getGeneratedRecordMappingFunction(bLangNode, symbolAtCursor, foundTypeLeft, foundTypeRight);
+                fEdits.add(new TextEdit(newFunctionRange, generatedRecordMappingFunction));
+            }
+        }
+        return fEdits;
+    }
+
+    /**
+     * Given two record types, this returns a function with mapped schemas.
+     *
+     * @param bLangNode      {@link BLangNode}
+     * @param symbolAtCursor {@link BSymbol}
+     * @param foundTypeLeft  {@link String}
+     * @param foundTypeRight {@link String}
+     * @return function string with mapped schemas
+     * @throws IOException throws if error occurred when getting mapped function
+     */
+    private static String getGeneratedRecordMappingFunction(BLangNode bLangNode, BSymbol symbolAtCursor,
+                                                            String foundTypeLeft, String foundTypeRight)
+            throws IOException {
+        JsonObject rightRecordJSON = new JsonObject();
+        JsonObject leftRecordJSON = new JsonObject();
+
+        // Schema 1
+        BType variableTypeMappingFrom = symbolAtCursor.type;
+        if (variableTypeMappingFrom instanceof BRecordType) {
+            List<BField> rightSchemaFields = new ArrayList<>(((BRecordType) variableTypeMappingFrom).fields.values());
+            JsonObject rightSchema = (JsonObject) recordToJSON(rightSchemaFields);
+
+            rightRecordJSON.addProperty("schema", foundTypeRight);
```

Comment:
Can we define set of constants for these fields as we are using them in multiple places. Then we can find the usages easily and change easily in future if requrired.

Reply:
Defined them as constants